state/cloudcredentials_test.go

```diff
@@ -111,7 +112,7 @@ func (s *CloudCredentialsSuite) TestUpdateCloudCredentialsExisting(c *gc.C) {
 	c.Assert(out, jc.DeepEquals, expected)
 }
 
-func (s *CloudCredentialsSuite) TestInvalidateCredential(c *gc.C) {
+func (s *CloudCredentialsSuite) assertCredentialInvalidated(c *gc.C, tag names.CloudCredentialTag) {
```

Comment:
is it possible to move the setup stuff into a different helper (setUpInvalidCredential) or something along those lines.
It is a bit confusing to have an:
assertCredentialInvalidated() that has a side effect of creating a cloud while
assertCredentialMarkedValid() only has the side effect of uploading the credential.
They seem like they should have similar side effects.
Then you would be able to do:
assertCredentialMarkedValid()
// and we can flip it back to invalid again
assertCredentialInvalidated()

Reply:
This whole area is not that simple - setting up of cloud, credential, etc....

For example, it is not possible to create an invalid credential - it makes sense to create valid credential, invalidate it and then use that in tests...

There are a few tests in this suite that are using credentials and, hence, create cloud, etc. As I am in this area, I expect I will circle these tests a few times and will strive for "Kaizen".

Landing as this, with a promise of follow-up improvements.

Reply:
$$merge$$